radiation/python/radsw/radsw_main.py

```diff
@@ -0,0 +1,4031 @@
+
+        for k in range(nlay):
+            tauray = colmol[k] * rayl
+
+            for j in range(NG28):
+                taur[k, NS28 + j] = tauray
+
+        for k in range(laytrop):
+            speccomb = colamt[k, 2] + self.strrat[12] * colamt[k, 5]
+            specmult = 8.0 * min(oneminus, colamt[k, 2] / speccomb)
+
+            js = 1 + int(specmult)
+            fs = np.mod(specmult, 1.0)
+            fs1 = 1.0 - fs
+            fac000 = fs1 * fac00[k]
+            fac010 = fs1 * fac10[k]
+            fac100 = fs * fac00[k]
+            fac110 = fs * fac10[k]
+            fac001 = fs1 * fac01[k]
+            fac011 = fs1 * fac11[k]
+            fac101 = fs * fac01[k]
+            fac111 = fs * fac11[k]
+
+            ind01 = id0[k, 27] + js
+            ind02 = ind01 + 1
+            ind03 = ind01 + 9
+            ind04 = ind01 + 10
+            ind11 = id1[k, 27] + js
+            ind12 = ind11 + 1
+            ind13 = ind11 + 9
+            ind14 = ind11 + 10
+
+            for j in range(NG28):
+                taug[k, NS28 + j] = speccomb * (
+                    fac000 * absa[ind01, j]
+                    + fac100 * absa[ind02, j]
+                    + fac010 * absa[ind03, j]
+                    + fac110 * absa[ind04, j]
+                    + fac001 * absa[ind11, j]
+                    + fac101 * absa[ind12, j]
+                    + fac011 * absa[ind13, j]
+                    + fac111 * absa[ind14, j]
+                )
+
+        for k in range(laytrop, nlay):
+            speccomb = colamt[k, 2] + self.strrat[12] * colamt[k, 5]
+            specmult = 4.0 * min(oneminus, colamt[k, 2] / speccomb)
+
+            js = 1 + int(specmult)
+            fs = np.mod(specmult, 1.0)
+            fs1 = 1.0 - fs
+            fac000 = fs1 * fac00[k]
+            fac010 = fs1 * fac10[k]
+            fac100 = fs * fac00[k]
+            fac110 = fs * fac10[k]
+            fac001 = fs1 * fac01[k]
+            fac011 = fs1 * fac11[k]
+            fac101 = fs * fac01[k]
+            fac111 = fs * fac11[k]
+
+            ind01 = id0[k, 27] + js
+            ind02 = ind01 + 1
+            ind03 = ind01 + 5
+            ind04 = ind01 + 6
+            ind11 = id1[k, 27] + js
+            ind12 = ind11 + 1
+            ind13 = ind11 + 5
+            ind14 = ind11 + 6
+
+            for j in range(NG28):
+                taug[k, NS28 + j] = speccomb * (
+                    fac000 * absb[ind01, j]
+                    + fac100 * absb[ind02, j]
+                    + fac010 * absb[ind03, j]
+                    + fac110 * absb[ind04, j]
+                    + fac001 * absb[ind11, j]
+                    + fac101 * absb[ind12, j]
+                    + fac011 * absb[ind13, j]
+                    + fac111 * absb[ind14, j]
+                )
+
+        return taug, taur
+
+    # The subroutine computes the optical depth in band 29:  820-2600
+    # cm-1 (low - h2o; high - co2)
+
+    def taumol29(
```

Comment:
these are different from the ones for the lw scheme?

Reply:
Yup, a whole bunch more of them haha